// @vuepress/core/lib/pluginApi/createPluginApiHooks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createPluginApiHooks = void 0;
const createHookQueue_1 = require("./createHookQueue");
const createPluginApiHooks = () => ({
    // life cycle hooks
    onInitialized: createHookQueue_1.createHookQueue('onInitialized'),
    onPrepared: createHookQueue_1.createHookQueue('onPrepared'),
    onWatched: createHookQueue_1.createHookQueue('onWatched'),
    onGenerated: createHookQueue_1.createHookQueue('onGenerated'),
    // page hooks
    extendsPageOptions: createHookQueue_1.createHookQueue('extendsPageOptions'),
    extendsPageData: createHookQueue_1.createHookQueue('extendsPageData'),
    // markdown hooks
    extendsMarkdown: createHookQueue_1.createHookQueue('extendsMarkdown'),
    // client files hooks
    clientAppEnhanceFiles: createHookQueue_1.createHookQueue('clientAppEnhanceFiles', true),
    clientAppRootComponentFiles: createHookQueue_1.createHookQueue('clientAppRootComponentFiles', true),
    clientAppSetupFiles: createHookQueue_1.createHookQueue('clientAppSetupFiles', true),
    // bundler hooks
    alias: createHookQueue_1.createHookQueue('alias'),
    define: createHookQueue_1.createHookQueue('define'),
});
exports.createPluginApiHooks = createPluginApiHooks;
